conan/tools/android/utils.py

```diff
@@ -0,0 +1,25 @@
+from conan.errors import ConanException
+
+
+def android_abi(conanfile, context="host"):
+    """
+    Returns Android-NDK ABI
+    :param conanfile: ConanFile instance
+    :param context: either "host", "build" or "target"
+    :return: Android-NDK ABI
+    """
+    if context not in ["host", "build", "target"]:
+        raise ConanException("context argument must be either 'host', 'build' or 'target'")
+
+    settings = getattr(conanfile, f"settings_{context}", conanfile.settings)
+    arch = settings.get_safe("arch")
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_ANDROID_ARCH_ABI.html
+    return {
+        "armv5el": "armeabi",
+        "armv5hf": "armeabi",
+        "armv5": "armeabi",
+        "armv6": "armeabi-v6",
+        "armv7": "armeabi-v7a",
+        "armv7hf": "armeabi-v7a",
+        "armv8": "arm64-v8a",
+    }.get(arch, arch)
```

Comment:
In the old code, `android_abi` would have been None for missing archs. The second parameter here changes that bahaviour, which might be more correct (No idea), but we should make sure it does not have any unintended side-effects

Reply:
Well there are 2 old codes, inconsistent:
- legacy `toos.to_android_abi()` => it fallbacks to `str(conanfile.settings.arch)`
- CMakeToolchain => fallbacks to None

So a decision has to be made for this new one, as you wish, I've not a strong opinion on this.

Reply:
I think falling back to the arch input instead of ``None`` might be better, for cases where the users extend ``settings.yml`` with their own architectures, if they map directly to android ones, then this will work. So I am fine with this as-is.